qiskit/aqua/components/neural_networks/quantum_generator.py

```diff
@@ -162,6 +162,22 @@ def set_discriminator(self, discriminator):
         """
         self._discriminator = discriminator
 
+    def set_optimizer(self, optimizer):
```

Comment:
Generally, elsewhere in Aqua, we use property setter/getters for setting/getting simple properties like this. I see though you have done it in a compatible way with set_seed and set_discriminator. Arguably I think they would be better off as changed to properties and the existing set_ methods deprecated.

For the optimizer setting, I think it should allow None to be set to revert back to the built in default - as is done in constructor with the default of None. Maybe the logic in the constructor should end up in a setter with the constructor  using the setter so the code for managing the None default and creating ADAM is done in the one place.

@Zoufalc Any thoughts/comments?

Reply:
Thanks for letting me know.

Does this sound right?
* Remove  `set_optimizer` and add a getter and setter (for the optimizer as well as the seed and discriminator), and move the default optimizer logic to the optimizer setter

In case I missed out on anything, or if you or @Zoufalc thinks there's anything else which I can add to the list, please  let me know!

Reply:
@woodsp-ibm Thanks for your feedback. That sounds very reasonable to me.

Reply:
That is what I had in mind. For set_seed and set_discriminator they would have to stay for compatibility but have deprecation messages if used telling people to use the setters.